.github/workflows/tests.yml

```
@@ -247,141 +325,152 @@ jobs:
-          # odd choice kept for parity with previous commit
-          - test-type: integration
-            default-channel: 'conda-forge'
-          - test-type: unit
-            default-channel: 'defaults'
+            test-group: 3
     env:
-      OS: macOS
-      PYTHON: ${{ matrix.python-version }}
-      TEST_SPLITS: ${{ matrix.test-splits }}
-      TEST_GROUP: ${{ matrix.test-group }}
+      PYTEST_MARKER: ${{ matrix.test-type == 'unit' && 'not integration' || 'integration' }}
+      PYTEST_SPLITS: ${{ matrix.test-type == 'unit' && '2' || '3' }}
+      REQUIREMENTS_TRUSTSTORE: ${{ contains('3.10|3.11|3.12', matrix.python-version) && '--file tests/requirements-truststore.txt' || '' }}
+
     steps:
-      - uses: actions/checkout@v4
+      - name: Checkout Source
+        uses: actions/checkout@v4
         with:
           fetch-depth: 0
 
-      - name: Cache conda packages
+      - name: Hash + Timestamp
+        run: echo "HASH=${{ runner.os }}-${{ runner.arch }}-Py${{ matrix.python-version }}-${{ matrix.default-channel }}-${{ matrix.test-type }}-${{ matrix.test-group }}-$(date -u "+%Y%m")" >> $GITHUB_ENV
+
+      - name: Cache Conda
         uses: actions/cache@v4
-        env:
-          # Increase this value to reset cache
-          CACHE_NUMBER: 4
         with:
           path: ~/conda_pkgs_dir
-          key:
-            ${{ matrix.arch }}-conda-${{ env.CACHE_NUMBER }}-py${{ env.PYTHON }}-${{ matrix.default-channel }}-${{
-            hashFiles('recipe/meta.yaml', 'dev/macos/setup.sh', 'tests/requirements.txt') }}
+          key: cache-${{ env.HASH }}
 
-      - uses: conda-incubator/setup-miniconda@v3
-        name: Setup miniconda for defaults
-        if: matrix.default-channel == 'defaults'
-        env:
-          CONDA_SOLVER: classic
+      - name: Setup Miniconda
+        uses: conda-incubator/setup-miniconda@v3
         with:
-          activate-environment: conda-test-env
-          miniconda-version: "latest"
+          condarc-file: .github/condarc-${{ matrix.default-channel }}
           python-version: ${{ matrix.python-version }}
-
-      - uses: conda-incubator/setup-miniconda@v3
-        name: Setup miniconda for conda-forge
-        if: matrix.default-channel == 'conda-forge'
-        env:
-          CONDA_SOLVER: classic
-        with:
-          activate-environment: conda-test-env
-          miniforge-version: latest
-          python-version: ${{ matrix.python-version }}
-
-      - name: Setup environment
-        shell: bash -l {0}
-        timeout-minutes: 15
-        run: |
-          ./dev/macos/setup.sh
-
-      - name: Python ${{ matrix.python-version }} on ${{ matrix.default-channel }}, ${{ matrix.test-type }} tests, group ${{ matrix.test-group }}
-        shell: bash -l {0}
-        run: |
-          ./dev/macos/${{ matrix.test-type }}.sh
-
-      - uses: codecov/codecov-action@v4
+          run-post: false  # skip post cleanup
+          miniconda-version: latest  # auto-install conda if missing
```

This will always install Miniconda, won't it? What about:

```suggestion
          miniconda-version: ${{ matrix.arch == 'osx-arm64' && 'latest' || null }}  # Miniconda not preinstalled in arm64 runners
```